app/src/main/java/programatorus/client/comm/app/proto/GetBoards.kt

```diff
@@ -0,0 +1,40 @@
+package programatorus.client.comm.app.proto
+
+import programatorus.client.comm.app.Requester
+import programus.proto.Protocol.GenericMessage
+import programus.proto.Protocol.GenericMessage.PayloadCase
+import programus.proto.Protocol.GetBoardsRequest
+
+class GetBoards : Requester<GetBoards.Boards> {
+
+    override fun prepareRequest(): GenericMessage.Builder =
+        GenericMessage.newBuilder()
+            .setGetBoardsRequest(GetBoardsRequest.newBuilder())
+
+    override val responsePayloadCase = PayloadCase.GETBOARDSRESPONSE
+
+    override fun handleResponse(message: GenericMessage): Boards {
+        val response = message.getBoardsResponse
+        return Boards(response.nameList.toTypedArray())
+    }
+
+    data class Boards(
```

Comment:
Boards DTO should contain also favorite boards. It can be just another array like this. It would need to have correct order. 
Alternatively it could be array of objects like `Pair<String, Int>`, where int would be index of the board. This would make it serialization-proof as array wouldn't need exact order.

Reply:
ok